162633497 Validate user sign up errors

jeanjoe opened the pull request:
**What does this PR do?**

This PR contains Unit Tests for User Sign up validation Errors.

**Description of Tasks completed**

* Add missing dependencies

* Add name to user sign up end point

* Add custom validation for password strength

* Add tests for user sign up errors

**How to manually run the test.**
To run the tests, use this command when virtual environment is activated

`python manage.py test`

**Background context**
The following files where modified to make the tests run successfully
* `.travis.yml`
* `authours/urls.py`
* `authors/apps/authentication/urls.py`
* `authors/apps/authentication/serializers.py`

**Relevant pivotal tracker story**

[DELIVERS #162633497]
(https://www.pivotaltracker.com/n/projects/2231917/stories/162633497)

jeanjoe committed:
feat(Validation): Descriptive user sign up validation errors

* Add missing dependencies

* Add name to user sign up end point

* Add custom validation for password strength

* Add tests for user sign up errors

[DELIVERS #162633497]
(https://www.pivotaltracker.com/n/projects/2231917/stories/162633497)

jeanjoe committed:
Ft(Validation): Descriptive user sign up validation errors

Fix travis command for running tests

chore(coveralls integration) integrate coveralls service

* Add .coveralls.yml file
Add coveralls file to enable integration with coveralls to attain the
code coverage service.
* Update requirements.txt with dependencies needed for code coverage service.
* Update readme with coveralls badge to get to code coverage
* Ensure that .travis.yml file has all the requirements needed to
* integrate with Travis.

* Add missing dependencies

* Add name to user sign up end point

* Add custom validation for password strength

* Add tests for user sign up errors

* Fix travis command for running tests

* change base test class name

chore(coveralls integration) integrate coveralls service

[Delivers #162633492](https://www.pivotaltracker.com/story/show/162633492)
[Delivers  #162633493](https://www.pivotaltracker.com/story/show/162633493)

mycok commented:
LGTM

mubstimor commented:
LGTM

Qadriyah commented:
LGTM

lym commented on authors/apps/authentication/serializers.py:
```diff
@@ -3,7 +3,20 @@
+        Test if password contains numbers and letters else raise serializer
+        Validation error
+        """
+        if (re.compile(r'^(?=.*[A-Za-z0-9])(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]').search(value['password'])) is None:
```
This line is too long. Consider refactoring it to:
```suggestion
       if re.compile(
            r'^(?=.*[A-Za-z0-9])(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]'
        ).search(value['password']) is None:
```

lym commented on authors/apps/authentication/serializers.py:
```diff
@@ -3,7 +3,20 @@
+        if (re.compile(r'^(?=.*[A-Za-z0-9])(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]').search(value['password'])) is None:
+            raise serializers.ValidationError(
+                {
+                    'password': "Password must contain a number, letters and a special character." 
```
This line is too long. Please fit it within 80 characters.

lym commented on authors/apps/authentication/tests/test_user_sign_up_validation_errors.py:
```diff
@@ -0,0 +1,97 @@
+        """
+        Test an if password length is less than 8 charaters.
+        """
+        passwordLenghtData = {
```
Please stick to [snake case](https://en.wikipedia.org/wiki/Snake_case) for all your _variable_ and _method_ names.

lym commented:
@jeanjoe Please address the comments and also squash all your commits into one.

lym commented on authors/apps/authentication/tests/base.py:
```diff
@@ -1,18 +1,25 @@
             }
         }
 
+    def createUser(self, data):
```
Use snake case here.

lym commented on authors/apps/authentication/tests/test_user_sign_up_validation_errors.py:
```diff
@@ -0,0 +1,104 @@
+        """
+        Test an if password length is less than 8 charaters.
+        """
+        password_lenght_data = {
```
Please correct the spelling of the word _length_ in the `password_lenght_data` variable name.

mubstimor committed:
chore(add db and deploy) Add PostgreSQL DB engine and deploy to Heroku

* Set default parameters for Postgresql database
* add settings for heroku deployment
* add db dependencies to repository.
* chore(code quality): Integrate Travis service
* add migrate scripts to Procfile to automate db migrations
* set allowed hosts to all
* add whitenoise to manage static files
* define travis test database configuration

[Delivers #162633492](https://www.pivotaltracker.com/story/show/162633492)
[Delivers  #162633493](https://www.pivotaltracker.com/story/show/162633493)

jeanjoe committed:
Merge branch 'develop' into ft-sign-up-validation-errors-162633497

jeanjoe committed:
Merge branch 'develop' into ft-sign-up-validation-errors-162633497
